specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+
+The secure monitor is responsible for context switching and isolation across domain boundaries, including event management. 
+
+For the purpose of this specification, TEE deployment models can be separated as:
+
+* Static partition TEE +
+A single TEE provides security services to Normal domain. TA are typically installed at boot by RoT FW and TEE OS, though Global Platform does also define protocols for installation of TA at runtime. System configuration and resource allocation can be mostly static, making the system more deterministic. +
+ +
+_Use case examples:_ edge devices and IoT, automation, and automotive. 
+* Virtualized TEE +
+On a virtualized system, TEE can also be virtualized. In this case a _secure partition manager_ in TEE domain is responsible for isolation of multiple TEE guests (for example, an OEM TEE and separate third party TEE). This model can also support more dynamic resource allocation. +
+ +
+_Use case examples:_ mobile clients, and automotive.
+
 ==== Isolation model
 
-==== Device access control
+A Global Platform TEE requires the following isolation guarantees:
 
-==== Sealing
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| Root domain MAY access resources assigned to any domain, but SHOULD prevent itself from unintended access to resources assigned to a different domain (privilege escalation).
+
+| CAT_NNN
+| No other domains can access resources assigned to Root domain
+
+| CAT_NNN
+| Resources assigned to TEE domain MUST NOT be accessible to Normal domain
+
+| CAT_NNN
+| Resources assigned to Normal domain MUST be accessible to Normal domain (r/w/x), and to TEE domain (r/w) (default sharing rule)
+
+| CAT_NNN
+| Resources assigned to a single TA, or a guest TEE, MUST not be accessible by a different TA, or guest TEE.
+
+|===
+
+The standard GP TEE model does not support sharing memory between TA in TEE domain. Each TA is expected to be a self-contained unit providing a specific security service, either to Normal domain or to other TA. All communications are implemented through secure channels managed by the TEE OS or SPM. 
+
+Processes in Normal domain can share memory assigned to Normal domain when interacting with a TA in TEE world (default sharing rule). Such shared memory can be cached when context switching between Normal and TEE domains.
+
+RISC-V hardware enforced isolation mechanisms can be used as follows to meet those guarantees:
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN 
+| PMP/ePMP, or MTT, MUST be used to isolate Root domain from other domains.
+
+| CAT_NNN  
+| Supervisor domains MUST be used to enforce isolation between Normal and TEE domains.
+
+|===
+
+See xref:chapter3.adoc#_supervisor_domains[supervisor domains].
+
+For static partition TEE, using PMP/ePMP or PMA with supervisor domains can be sufficient. 
+
```

Comment:
PMA is not a isolation mechanism. Should it have been MTT?

Reply:
Why is PMA not an isolation mechanism? A memory map that doesn't map a portion of the physical address space that contains a secure device would be isolating it.

Reply:
PMA specifies attributes for memory - e.g. uncacheable, it does not specify access permissions.

Reply:
There is nothing preventing PMA to be used for this. They go hand in hand with PMP to enforce memory access rules. Granted there are no standardised fields for it, but it still an implementation choice available "underneath" smmtt. 

Arguably, MTT combines PMP with PMA (svpam) in the same way.

Reply:
then perhaps state it this way:

"For static partition TEE, using PMP/ePMP, or MTT, with supervisor domains can be sufficient - with PMAs used for to assign additional attributes to physical memory e.g. cryptographic memory protection."

Reply:
I have re-written this. See if that works better.